src/utils/ServerSocket.js

```diff
@@ -125,4 +152,16 @@ export default class ServerSocket {
     off(eventName, handler) {
         this._socket.off(eventName, handler);
     }
+
+    /**
+     * Intercept event and let ReconnectManager handle them.
+     */
+    _interceptEvent(eventName, handler) {
+        if (!this._reconnectManager) throw new Error("ReconnectMangager is missing");
+
+        this._reconnectManager.on(eventName, handler);
```

Comment:
If we are going to throw an error we should have some error handling in the caller IMO. Instead, I would simply remove this and replace with a true check instead:

```suggestion
        if(this._reconnectManager) {
            this._reconnectManager.on(eventName, handler);
        }

```